WMIAdventure/frontend/src/js/storage/effects/effectStorage.js

```diff
@@ -0,0 +1,58 @@
+import CardsAPIGateway from "../../api/gateways/CardsAPIGateway";
+import {get, getWithSetCallback, set} from "../cache/cache";
+import {effectKeys} from "../localStorageKeys";
+
+/**
+ * Returns list of all effects.
+ * @returns {Promise<[]>}
+ */
+export const getAllEffects = async () => {
+    let allEffects = [];
+    let effectsIds = get(effectKeys.effectsIds);
+
+    async function retrieveEffectsFromAPIAndSaveToStorage() {
+        allEffects = await CardsAPIGateway.getAllEffects();
+
+        // We save array of effects ids in storage. Later we can iterate over this array and retrieve individual effects.
+        effectsIds = allEffects.map((effect) => effect.id);
+        set(effectKeys.effectsIds, effectsIds);
+
+        for (const effect of allEffects) {
+            set(effectKeys.effectKey(effect.id), effect);
```

Comment:
Dlaczego przechowujemy osobno ID efektu i cały efekt (ID, nazwa, ikona)? To osobne ID jest nam po coś potrzebne?

Reply:
Myślałem, że szkoda miejsca na przechowywanie całego jsona wszystkich efektów. Ta lista idków efektów służy do tego, że jak chcemy wszystkie efekty wyciągnąć ze storage to sobie możemy przeiterować przez tą liste z id-kami i wyciągać efekty po kolei. A z kolei jak chcemy konkretny efekt wyciągnąć to po od razu wyciągamy konkretny efekt ze storage, nie musimy wyciągać całego jsona z wszystkimi efektami i go przeszukiwać w poszukiwaniu konkretnego efektu

Reply:
Wg mnie to można na luzie przechować całość. Tych efektów jest mało i to nie zajmie praktycznie nic, a potem nie musielibyśmy lazily pobierać danych przy walce, tylko już by tam były

Reply:
Mogę zmienić, ale nie do końca rozumiem co rozumiesz przez lazily, jak te efekty wszystkie są zapisywane do storage w tym miejscu właśnie

Reply:
Chodzi o to, że w storage są efekty pojedynczo, a nie ma jednego jsona, co przechowuje wszystkie efekty naraz?

Reply:
Aa jasne, teraz rozumiem, to faktycznie tak jest dużo lepiej

Reply:
oks, to może gdzieś komentarz dodać, bo to troche nieintuicyjne?